docs/interfaces.rst

```diff
@@ -0,0 +1,179 @@
+
+.. code-block:: python
+
+    interface FooBar:
+        def calculate() -> uint256: view
+        def test1(): nonpayable
+
+The defined interface can then be use to make external calls, given a contract address:
+
+.. code-block:: python
+
+    @external
+    def test(some_address: address):
+        FooBar(some_address).calculate()
+
+The interface name can also be used as a type annotation for storage variables. You then assign an address value to the variable to access that interface. Note that assignment of an address requires the value to be cast using the interface type e.g. ``FooBar(<address_var>)``:
+
+.. code-block:: python
+
+    foobar_contract: FooBar
+
+    @external
+    def __init__(foobar_address: address):
+        self.foobar_contract = FooBar(foobar_address)
+
+    @external
+    def test():
+        self.foobar_contract.calculate()
+
+Specifying ``payable`` or ``nonpayable`` annotation indicates that the call made to the external contract will be able to alter storage, whereas the ``view`` ``pure`` call will use a ``STATICCALL`` ensuring no storage can be altered during execution. Additionally, ``payable`` allows non-zero value to be sent along with the call.
+
+.. code-block:: python
+
+    interface FooBar:
+        def calculate() -> uint256: pure
+        def query() -> uint256: view
+        def update(): nonpayable
+        def pay(): payable
+
+    @external
+    def test(some_address: address):
+        FooBar(some_address).calculate()  # cannot change storage
+        FooBar(some_address).query()  # cannot change storage, but reads itself
+        FooBar(some_address).update()  # storage can be altered
+        FooBar(some_address).pay(value=1)  # storage can be altered, and value can be sent
+
+Importing Interfaces
+====================
+
+Interfaces are imported with ``import`` or ``from ... import`` statements.
+
+Imported interfaces are written using standard Vyper syntax, with the body of each function replaced by a ``pass`` statement:
+
+.. code-block:: python
+
+    @external
+    def test1():
+        pass
+
+    @external
+    def calculate() -> uint256:
+        pass
+
+You can also import a fully implemented contract and Vyper will automatically convert it to an interface. It is even possible for a contract to import itself to gain access to it's own interface.
+
+Imports via ``import``
+----------------------
+
+With absolute ``import`` statements, you **must** include an alias as a name for the imported package. In the following example, failing to include ``as Foo`` will raise a compile error:
+
+.. code-block:: python
+
+    import contract.foo as Foo
+
+Imports via ``from ... import``
+-------------------------------
+
+Using ``from`` you can perform both absolute and relative imports. With ``from`` import statements you **cannot** use an alias - the name of the interface will always be that of the file:
+
+.. code-block:: python
+
+    from contract import foo
```

Comment:
These two import rules seem weird together

Reply:
Yeah, this is what #2083 addresses

Reply:
So #2083 would make it so you can use an alias on `from ...` statements, right?

Does it also allow _not_ using an alias on regular imports? I feel like this would be most consistent

Reply:
It doesn't work very well for regular imports, because you'd end up with an interface named `path.to.interface`

Reply:
> It doesn't work very well for regular imports, because you'd end up with an interface named `path.to.interface`

Ahhhhh, gotcha